endpoint/endpoint_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"testing"
+)
+
+func TestOwner(t *testing.T) {
+	e := &Endpoint{}
+	if e.Owner() != "" {
+		t.Errorf("Owner should be empty if Labels not initialized")
+	}
+	e.Labels = map[string]string{}
+	e.Labels[ownerKey] = ""
+	if e.Owner() != "" {
+		t.Errorf("Owner should be empty for records without owners")
+	}
+	e.Labels[ownerKey] = "owner"
+	if e.Owner() != "owner" {
+		t.Errorf("Incorrect owner is returned")
+	}
+}
+
+func TestSetOwner(t *testing.T) {
+	e := &Endpoint{}
+	e.SetOwner("")
+	if e.Labels == nil {
+		t.Fatalf("Labels map should be initialized")
+	}
+	if e.Labels[ownerKey] != "" {
+		t.Errorf("Owner set incorrectly")
+	}
+	e.SetOwner("owner")
+	if e.Labels[ownerKey] != "owner" {
+		t.Errorf("Owner set incorrectly")
+	}
+}
```

Comment:
Just a thought: We could introduce a `NewEndpoint()` function which initializes the Labels map. I think we'll do that anyway at some point. Then, since the default for a missing key would `""` the code would become quite less verbose, I think. I'd then probably just use `endpoint.Labels["owner"]` everywhere instead of separate methods. However, just a different style. If you like it more this way, I'm fine.